datafusion/core/src/physical_optimizer/pipeline_fixer.rs

```diff
@@ -49,16 +57,18 @@ impl PipelineFixer {
     }
 }
 type PipelineFixerSubrule =
-    dyn Fn(&PipelineStatePropagator) -> Option<Result<PipelineStatePropagator>>;
+    dyn Fn(PipelineStatePropagator) -> Option<Result<PipelineStatePropagator>>;
```

Comment:
Could you please add some comments for the type definition `dyn Fn(PipelineStatePropagator) -> Option<Result<PipelineStatePropagator>>` ?

Reply:
Sure, will do.